gravitee-apim-console-webui/src/management/integrations/create-integration/create-integration.component.spec.ts

```diff
@@ -0,0 +1,125 @@
+
+import { CreateIntegrationComponent } from './create-integration.component';
+import { CreateIntegrationHarness } from './create-integration.harness';
+
+import { IntegrationsModule } from '../integrations.module';
+import { CONSTANTS_TESTING, GioTestingModule } from '../../../shared/testing';
+import { SnackBarService } from '../../../services-ngx/snack-bar.service';
+import { CreateIntegrationPayload } from '../integrations.model';
+
+describe('CreateIntegrationComponent', () => {
+  let fixture: ComponentFixture<CreateIntegrationComponent>;
+  let componentHarness: CreateIntegrationHarness;
+  let httpTestingController: HttpTestingController;
+
+  const fakeSnackBarService = {
+    error: jest.fn(),
+  };
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      declarations: [CreateIntegrationComponent],
+      imports: [GioTestingModule, IntegrationsModule, BrowserAnimationsModule, NoopAnimationsModule],
+      providers: [
+        {
+          provide: SnackBarService,
+          useValue: fakeSnackBarService,
+        },
+      ],
+    })
+      .overrideProvider(InteractivityChecker, {
+        useValue: {
+          isFocusable: () => true, // This traps focus checks and so avoid warnings when dealing with
+          isTabbable: () => true, // This traps focus checks and so avoid warnings when dealing with
+        },
+      })
+      .compileComponents();
+  });
+
+  beforeEach(async () => {
+    fixture = TestBed.createComponent(CreateIntegrationComponent);
+    httpTestingController = TestBed.inject(HttpTestingController);
+    componentHarness = await TestbedHarnessEnvironment.harnessForFixture(fixture, CreateIntegrationHarness);
+    fixture.detectChanges();
+  });
+
+  afterEach(() => {
+    httpTestingController.verify();
+  });
+
+  describe('form', () => {
+    it('should not submit empty form', async () => {
+      await componentHarness.clickOnSubmit();
+      httpTestingController.expectNone(`${CONSTANTS_TESTING.env.v2BaseURL}/integrations`);
+    });
+
+    it('should create integration with valid form', async () => {
+      const expectedPayload: CreateIntegrationPayload = {
+        name: 'TEST123',
+        description: '',
+        provider: 'AWS',
+      };
+      await componentHarness.setName('TEST123');
+      await componentHarness.clickOnSubmit();
+      expectIntegrationPostRequest(expectedPayload);
+    });
+
+    it('should create integration with description', async () => {
+      const expectedPayload: CreateIntegrationPayload = {
+        name: 'TEST123',
+        description: 'Test Description',
+        provider: 'AWS',
+      };
+      await componentHarness.setName('TEST123');
+      await componentHarness.setDescription('Test Description');
+      await componentHarness.clickOnSubmit();
+
+      expectIntegrationPostRequest(expectedPayload);
+    });
+
+    it('should handle error with message', async () => {
```

Comment:
We should also test the form validation:
- name too short
- name too long
- description too long

Reply:
tested